staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -338,15 +338,17 @@ func Start(ctx context.Context, nodeServer interface{}, opts ...Option) (result
 	// Run the node plugin gRPC server first to ensure that it is ready.
 	var supportedServices []string
 	plugin, err := startGRPCServer(klog.NewContext(ctx, klog.LoggerWithName(logger, "dra")), o.grpcVerbosity, o.unaryInterceptors, o.streamInterceptors, o.draEndpoint, func(grpcServer *grpc.Server) {
-		if nodeServer, ok := nodeServer.(drapbv1alpha4.NodeServer); ok && o.nodeV1alpha4 {
-			logger.V(5).Info("registering v1alpha4.Node gGRPC service")
-			drapbv1alpha4.RegisterNodeServer(grpcServer, nodeServer)
-			supportedServices = append(supportedServices, drapbv1alpha4.NodeService)
-		}
-		if nodeServer, ok := nodeServer.(drapbv1beta1.DRAPluginServer); ok && o.nodeV1beta1 {
-			logger.V(5).Info("registering v1beta1.DRAPlugin gRPC service")
-			drapbv1beta1.RegisterDRAPluginServer(grpcServer, nodeServer)
-			supportedServices = append(supportedServices, drapbv1beta1.DRAPluginService)
+		for _, nodeServer := range nodeServers {
```

Comment:
do we want to provide priority or left to the developer to pass the order of preference of the version they prefer?

Reply:
The code registers everything that is passed in and enabled via options. It doesn't matter which server gets registered first (alpha or beta).

Theoretically, a driver could pass an implementation for the same version twice. I haven't tested it, but I suppose registration then will barf or one server overwrite the other.

Reply:
no worries, was just a curiosity question